src/logging.cpp

```diff
@@ -46,9 +47,10 @@ void log(verbosity v, char const* tag, char const* msg, va_list args)
         va_end(copy);
         return;
     }
-    char buf[512];
-    char* out = buf;
-    if (sz >= static_cast<int>(sizeof(buf))) {
+    constexpr auto max_stack_buffer_size = 512;
+    std::array<char, max_stack_buffer_size> buf;
+    char* out = buf.data();
+    if (sz >= static_cast<int>(buf.size())) {
```

Comment:
Per clang-tidy:
1. avoid C-style arrays
1. avoid magic numbers.

Reply:
Thought, `max_stack_buffer_size` becomes `size_t`, and:
```c
if (sz >= max_stack_buffer_size) {
    out = new char[sz];
} else {
    out = buf.data();
}
```

Reply:
yeah that would be better. I'll change it. Thanks